crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -1,17 +1,70 @@
 use miden_objects::{
     assembly::{ProcedureName, QualifiedProcedureName},
     note::{Note, NoteScript},
+    utils::{sync::LazyLock, Deserializable},
+    vm::Program,
     Digest,
 };
 
-use crate::{
-    account::{
-        components::basic_wallet_library,
-        interface::{AccountComponentInterface, AccountInterface},
-    },
-    note::scripts::{p2id, p2id_commitment, p2idr, p2idr_commitment, swap, swap_commitment},
+use crate::account::{
+    components::basic_wallet_library,
+    interface::{AccountComponentInterface, AccountInterface},
 };
 
+// WELL KNOWN NOTE SCRIPTS
+// ================================================================================================
+
+// Initialize the P2ID note script only once
+static P2ID_SCRIPT: LazyLock<NoteScript> = LazyLock::new(|| {
+    let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/note_scripts/P2ID.masb"));
+    let program = Program::read_from_bytes(bytes).expect("Shipped P2ID script is well-formed");
+    NoteScript::new(program)
+});
+
+// Initialize the P2IDR note script only once
+static P2IDR_SCRIPT: LazyLock<NoteScript> = LazyLock::new(|| {
+    let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/note_scripts/P2IDR.masb"));
+    let program = Program::read_from_bytes(bytes).expect("Shipped P2IDR script is well-formed");
+    NoteScript::new(program)
+});
+
+// Initialize the SWAP note script only once
+static SWAP_SCRIPT: LazyLock<NoteScript> = LazyLock::new(|| {
+    let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/note_scripts/SWAP.masb"));
+    let program = Program::read_from_bytes(bytes).expect("Shipped SWAP script is well-formed");
+    NoteScript::new(program)
+});
+
+/// Returns the P2ID (Pay-to-ID) note script.
+pub fn p2id() -> NoteScript {
+    P2ID_SCRIPT.clone()
+}
+
+/// Returns the P2ID (Pay-to-ID) note script commitment.
+pub fn p2id_commitment() -> Digest {
+    P2ID_SCRIPT.hash()
+}
```

Comment:
Why do these need to be public? Can we not get the same info from the `WellKnownNote` enum?